src/run_bpftrace.cpp

```diff
@@ -34,10 +34,34 @@ void check_is_root()
   }
 }
 
-int run_bpftrace(BPFtrace &bpftrace, Output &output, BpfBytecode &bytecode)
+int run_bpftrace(BPFtrace &bpftrace,
+                 Output &output,
+                 BpfBytecode &bytecode,
+                 std::unordered_map<std::string, std::string> &&named_params,
```

Comment:
Why not just push down the parsing into this layer? Just pass the remaining arg pointer (or a std::vector<args> that are unprocessed). Then you can reduce duplication with the AOT case, which essentially has the same code right now.

Reply:
Because AOT and main parse the raw args slightly differently e.g. there is no `--` needed for AOT binaries - everything is in the same arg namespace.

Reply:
> or a std::vector that are unprocessed

Ok, this is what I did. Good call.